### PR TITLE
Make portal more modular

### PR DESCRIPTION
**Pull request by TwistedAsylumMC**
Portal currently uses a lot of global variables and singletons, a lot more than it should. This pull request intends to make portal use no global variables and make it more modular to create a better API for users.

### CODE REVIEW

**Comment by JustTalDevelops** [comment]
finish smh

File: session/store.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	s.sessions[x.UUID()] = x
+	s.sessionNames[x.Conn().IdentityData().DisplayName] = x
+}
+
+// Delete deletes a session from the store.
+func (s *Store) Delete(x uuid.UUID) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	v, ok := s.sessions[x]
```
[the remark this answers is not in the document]
Comment (TwistedAsylumMC) [reply session/store.go:66]: Not exactly, woulc cause nil pointer errors a few lines later when it deletes from the sessionNames map

[the remark this answers is not in the document]
Comment (T14Raptor) [reply session/store.go:66]: Never mind I see what you’re saying

File: socket/server.go [hunk 2]
```diff
@@ -77,7 +88,10 @@ func (s *DefaultServer) Listen() error {
 
 // handleClient handles a client that has been accepted from the server.
 func (s *DefaultServer) handleClient(c *Client) {
-	defer c.Close(s.serverRegistry)
+	defer s.handleClientDisconnect(c)
+	s.clientsMu.Lock()
+	s.clients[c.Name()] = c
```
Comment (Unickorn) [inline socket/server.go:93]: At this stage, the client hasn't sent any packet other than initializing a TCP connection, so the name is empty. It'd make sense to move this to the auth request handler.

**Review by T14Raptor — APPROVED** [review]

**Review by DaPigGuy — APPROVED** [review]

**Pull request merged**